### PR TITLE
fix(ci): Only bumpdep Halyard for master branch

### PR DESCRIPTION
**Pull request by kskewes-sf**
Halyard releases are different and release-* branches have no connection
with Spinnaker service releases.

We only want to bumpdep Halyard if tag was pushed to `master` branch.

For services we always want to bumpdep same branch.

### CODE REVIEW

File: .github/workflows/bump_dependencies.yml [hunk 1]
```diff
@@ -9,12 +9,23 @@ jobs:
     if: startsWith(github.repository, 'spinnaker/')
     runs-on: ubuntu-latest
     steps:
+    # Halyard releases are separate from Spinnaker releases so we need to
+    # exclude Halyard from bump deps when targeting services release-* branches.
+    - name: decide bumpdep target repositories
+      id: bumpdep_targets
+      run: |
+        REPOS=clouddriver,echo,front50,gate,halyard,igor,keel,orca
+        if [ "${{ github.event.client_payload.branch }}" != "  origin/master"]; then
```
Comment (kskewes-sf) [inline .github/workflows/bump_dependencies.yml:18]: Based on [this GHA run](https://github.com/spinnaker/fiat/runs/5951468606?check_suite_focus=true):
```
Run BRANCHES=$(git branch -r --contains refs/tags/v1.31.0)
BRANCHES is '  origin/master
  origin/release-1.28.x'
NUM_BRANCHES is '2'
exactly one branch required to release fiat, but there are 2 (  origin/master
  origin/release-1.28.x)
```

File: .github/workflows/bump_dependencies.yml [hunk 2]
```diff
@@ -9,12 +9,23 @@ jobs:
     if: startsWith(github.repository, 'spinnaker/')
     runs-on: ubuntu-latest
     steps:
+    # Halyard releases are separate from Spinnaker releases so we need to
+    # exclude Halyard from bump deps when targeting services release-* branches.
+    - name: decide bumpdep target repositories
+      id: bumpdep_targets
+      run: |
```
Comment (kskewes-sf) [inline .github/workflows/bump_dependencies.yml:16]:
```
$ REPOS=clouddriver,echo,front50,gate,igor,keel,orca
$ test="  origin/master"
$ if [ "$test" == "  origin/master" ]; then REPOS+=',halyard'; fi

# note that the leading spaces get trimmed
$ echo $test
origin/master

$ echo $REPOS
clouddriver,echo,front50,gate,igor,keel,orca,halyard

# test with no leading space
$ if [ "$test" == "origin/master" ]; then REPOS+=',noleadingspaceincondition'; fi

$ echo $REPOS
clouddriver,echo,front50,gate,igor,keel,orca,halyard
```

**Comment by dbyron-sf** [comment]
@mergifyio update

File: .github/workflows/bump_dependencies.yml [hunk 3]
```diff
@@ -9,12 +9,23 @@ jobs:
...
+        REPOS=clouddriver,echo,front50,gate,igor,keel,orca
+        if [ "${{ github.event.client_payload.branch }}" == "  origin/master"]; then
```
Comment (dbyron-sf) [inline .github/workflows/bump_dependencies.yml:18]: The whitespace here rubs me the wrong way, but that's for another day.

**Review by dbyron-sf — APPROVED** [review]

**Comment by dbyron-sf** [comment]
@mergifyio backport release-1.27.x release-1.28.x

File: .github/workflows/bump_dependencies.yml (hunk 3 above)
Comment (kskewes-sf) [reply .github/workflows/bump_dependencies.yml:18]: Cheers. Me too. For some reason that's how it comes through. We could regex match instead but I went with simpler option.

**Pull request merged**